### PR TITLE
Fix #29 - Turn ShapeType into an enum

### LINKED ISSUE #29: Change the ShapeType class to be an enum
The `PhpMyAdmin\ShapeFile\ShapeType` class should be changed to a PHP backed enum and the ShapeType constants should be turned into enum cases.

Currently, the ShapeFile class accepts any integer as a shape type. Only the shape type defined in the ShapeType class should be accepted.

- Related to #26

### PR DESCRIPTION
**Pull request by JoolsMcFly**
Hi!

Thought I'd give a go at #29 .

In various places `shapeType` was set to `-1` as default value. I replaced it with `ShapeType::NULL` and would like feed back on that.

Thanks for reviewing!

- Fixes #29

- Related to https://github.com/phpmyadmin/shapefile/pull/26

### CODE REVIEW

**Comment by JoolsMcFly** [comment]
Hi. Is there anything I can do on my side to get this merged?
Thanks!

**Review by williamdes — APPROVED** [review]
Thank you for this improvement !

**Comment by williamdes** [comment]
What do you think @MauricioFauth ?

**Review by MauricioFauth — APPROVED** [review]

File: src/ShapeFile.php
Comment (MauricioFauth) [inline src/ShapeFile.php]: I think it's better to add an `Unknown` case instead of using `Null`, because `Null` is a valid type.
```suggestion
        $this->shapeType = ShapeType::tryFrom((int) $shapeType) ?? ShapeType::Unknown;
```

File: src/ShapeFile.php
Comment (MauricioFauth) [inline src/ShapeFile.php]: It shouldn't break if it's a Null shape.

File: src/ShapeType.php
Comment (MauricioFauth) [inline src/ShapeType.php]: Enum case names uses the PascalCase format.

```suggestion
    case PolyLine = 3;
```

File: tests/ShapeFileTest.php
Comment (MauricioFauth) [inline tests/ShapeFileTest.php]:
```suggestion
        $obj = new ShapeRecord(ShapeType::Unknown);
        $this->assertEquals('Unknown Shape', $obj->getShapeName());
```

**Review by MauricioFauth — CHANGES_REQUESTED** [review]

**Comment by JoolsMcFly** [comment]
Hi @MauricioFauth . Thanks for your feedback.

I just added a commit which turns enum cases in PascalCase and adds an Unknown type.

Lemme know what you think.

Thanks!

File: src/ShapeFile.php
Comment (MauricioFauth) [inline src/ShapeFile.php]: You should keep the strict check for `false`, because type casting `false` to `int` will convert it to `0`, which means `ShapeType::Null` and not `ShapeType::Unknown`.

File: tests/ShapeFileTest.php
Comment (MauricioFauth) [inline tests/ShapeFileTest.php]: You can pass a enum directly instead of an int.
```suggestion
    public function testShapeSaveLoad(ShapeType $shapeType, array $points): void
    {
```

File: tests/ShapeFileTest.php
Comment (MauricioFauth) [inline tests/ShapeFileTest.php]: You can pass a enum directly instead of an int.
```suggestion
            [ShapeType::Point, $pointsForPointType],
```

File: tests/ShapeFileTest.php
Comment (MauricioFauth) [inline tests/ShapeFileTest.php:307]: You should also add tests for Unknown case as well.

**Review by MauricioFauth — CHANGES_REQUESTED** [review]

**Comment by MauricioFauth** [comment]
@JoolsMcFly Could you please add a `Signed-off-by` line to the commit message using `git commit --signoff` or `git commit -s`?

**Comment by JoolsMcFly** [comment]
> @JoolsMcFly Could you please add a `Signed-off-by` line to the commit message using `git commit --signoff` or `git commit -s`?

Hi @MauricioFauth .
If you don't mind I'll wait until I get green light before I squash my commits and sign. Is that OK?

**Comment by JoolsMcFly** [comment]
Thanks for your review. It all makes sense and I've made required changes.

**Review by MauricioFauth — APPROVED** [review]
Great work! Thanks!

**Comment by JoolsMcFly** [comment]
Alright, squashed and signed!

**Pull request merged**
